### PR TITLE
cache with concatenated string of input function as key

### PR DESCRIPTION
**Pull request by ketanbj**
tSache.h - added LRU cache implementation with concatenated string as key for a map based LRU cache

### CODE REVIEW

**Review by davidbrownell — CHANGES_REQUESTED** [review]
If this is code used to test an idea, it shouldn't be checked into main. If this is code that improves performance, please delete the commented-out code.

**Comment by ketanbj** [comment]
This was to test out the idea of using concatenated string as key which unfortunately does not improve performance. However, this repository is our own version of lru benchmark. I thought it would be good to have this for posterity purposes and may be it is useful in some other condition. Your call - we can keep that in this branch itself and not merge it.

**Comment by davidbrownell** [comment]
> This was to test out the idea of using concatenated string as key which unfortunately does not improve performance. However, this repository is our own version of lru benchmark. I thought it would be good to have this for posterity purposes and may be it is useful in some other condition. Your call - we can keep that in this branch itself and not merge it.

Yeah, I'd prefer that code in main is production quality (for all of our repos).

**Pull request merged**
